### PR TITLE
feat: support atproto source and bluesky worker

### PR DESCRIPTION
**Pull request by polebug**
## Checklist

- [X] The commit message follows [Angular Contributing guidelines](https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit);
- [X] Tests for the changes have been added (for bug fixes / features);

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: provider/atproto/bluesky/client.go [hunk 1]
```diff
@@ -0,0 +1,597 @@
...
+type Client struct {
+	username       string
+	password       string
+	filter         []string
+	timestampStart int64
+	mutex          sync.RWMutex
+	defaultClient  *XrpcClient
+	cacheClient    map[string]*XrpcClient
+	encoder        *form.Encoder
+	httpClient     *http.Client
```
Comment (brucexc) [inline provider/atproto/bluesky/client.go:44]: These parameters are not used in the code.

File: provider/atproto/bluesky/client.go [hunk 2]
```diff
@@ -0,0 +1,597 @@
...
+		Did:        auth.Did,
+	}
+
+	return &XrpcClient{
+		Client:    client,
+		createdAt: time.Now(),
+	}, nil
+}
+
+func (c *Client) LookupDIDEdnpoint(ctx context.Context, did syntax.DID) string {
```
Comment (brucexc) [inline provider/atproto/bluesky/client.go:550]: typo

File: provider/atproto/bluesky/client.go [hunk 3]
```diff
@@ -0,0 +1,597 @@
...
+	c.cacheClient[endpoint] = client
+	c.mutex.Unlock()
+
+	return client, nil
+}
+
+func (c *Client) createAndAuthenticateClient(ctx context.Context, endpoint string) (*XrpcClient, error) {
+	client := &xrpc.Client{
+		Host:   endpoint,
+		Client: http.DefaultClient,
```
Comment (brucexc) [inline provider/atproto/bluesky/client.go:520]: Use default Client in xrpc.Client

File: internal/engine/worker/atproto/core/worker.go [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+}
+
+// buildPostMetadata constructs metadata for a post message.
+func (w *worker) buildPostMetadata(message at.Message) *metadata.SocialPost {
+	post := &metadata.SocialPost{
+		Handle:        message.Handle,
+		ProfileID:     message.Did.String(),
+		PublicationID: message.Rkey,
+		ContentURI:    message.URI,
+		Timestamp:     uint64(message.CreatedAt.Unix()),
```
Comment (brucexc) [inline internal/engine/worker/atproto/core/worker.go:191]: When message.CreatedAt is set to 0001-01-01 00:00:00 +0000, it should default to using message.Feed.CreatedAt

File: internal/engine/worker/atproto/core/worker.go [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+	}
+}
+
+func (w *worker) Types() []schema.Type {
+	return []schema.Type{
+		typex.SocialComment,
+		typex.SocialPost,
+		typex.SocialShare,
+		typex.SocialProfile,
+	}
```
Comment (brucexc) [inline internal/engine/worker/atproto/core/worker.go:60]: missing typex.SocialLike

File: internal/engine/worker/atproto/core/worker.go (hunk 4 above)
Comment (polebug) [reply internal/engine/worker/atproto/core/worker.go:192]: `message.CreatedAt` is the result of parsing `message.Feed.CreatedAt`

File: config/config.go [hunk 6]
```diff
@@ -90,6 +90,7 @@ type Server struct {
 type Component struct {
 	RSS           *Module   `mapstructure:"rss"`
 	Federated     []*Module `mapstructure:"federated" validate:"dive"`
+	Atproto       []*Module `mapstructure:"atproto" validate:"dive"`
```
Comment (brucexc) [inline config/config.go:93]: redundant field

File: cmd/main.go [hunk 7]
```diff
@@ -324,10 +324,16 @@ func findModuleByID(configFile *config.File, workerID string) (*config.Module, e
...
 
+	// Search in RSS components
 	if module, found := findInComponent([]*config.Module{configFile.Component.RSS}); found {
 		return module, nil
 	}
 
+	// Search in atproto components
+	if module, found := findInComponent(configFile.Component.Atproto); found {
+		return module, nil
+	}
```
Comment (brucexc) [inline cmd/main.go:335]: redundant checks

File: deploy/config.example.yaml [hunk 8]
```diff
@@ -85,13 +85,21 @@ component:
...
         relay_url_list: [ "https://relay.fedi.buzz/instance/mastodon.social"]
         port: 8181
+    # bluesky
+    - id: bluesky-core
+      network: bluesky
+      worker: bluesky
+      endpoint:
+      parameters:
+        username:
+        password:
```
Comment (brucexc) [inline deploy/config.example.yaml:105]: id format to network-worker

File: internal/node/component/info/network_config.go [hunk 9]
```diff
@@ -451,7 +451,7 @@ var NetworkToWorkersMap = map[network.Network][]worker.Worker{
 // WorkerToConfigMap is a map of worker to config.
 var WorkerToConfigMap = map[network.Protocol]map[worker.Worker]workerConfig{
 	network.ActivityPubProtocol: {
```
Comment (brucexc) [inline internal/node/component/info/network_config.go:453]: add the configuration infi related to bluesky

File: internal/database/dialer/postgres/client.go [hunk 10]
```diff
@@ -487,6 +479,79 @@ func (c *client) GetUpdatedMastodonHandles(ctx context.Context, query model.Quer
...
+	return c.database.WithContext(ctx).Clauses(onConflictClause).CreateInBatches(&values, math.MaxUint8).Error
+}
+
+func (c *client) LoadDatasetBlueskyProfiles(ctx context.Context, query model.QueryBlueskyProfiles) ([]*model.BlueskyProfile, error) {
+	databaseStatement := c.database.WithContext(ctx).Table(table.DatasetBlueskyProfile{}.TableName())
+
+	if query.Cursor != nil {
+		var cursor *table.DatasetBlueskyProfile
+
+		if err := c.database.WithContext(ctx).First(&cursor, "did = ?", query.Cursor).Error; err != nil {
```
Comment (brucexc) [inline internal/database/dialer/postgres/client.go:510]: did -> handle

**Review by brucexc — APPROVED** [review]

**Review by pseudoyu — APPROVED** [review]

**Pull request merged**